File: apps/desktop/src-tauri/src/events.rs
pub const EVENT_MAIN_WINDOW_TOGGLE: &str = "main-window-toggle";
pub const EVENT_HOTKEY_NAVIGATE: &str = "hotkey-navigate";
pub const EVENT_CLIPBOARD_CHANGED: &str = "clipboard-changed";
pub const EVENT_TODO_REMINDER_FIRED: &str = "todo-reminder-fired";
pub const EVENT_GLOBAL_NOTIFICATION_PUSH: &str = "global-notification-push";
pub const EVENT_POMODORO_STATE_CHANGED: &str = "pomodoro-state-changed";
pub const EVENT_QUICK_CAPTURE_RESET: &str = "quick-capture-reset";
pub const EVENT_SPOTLIGHT_RESET: &str = "spotlight-reset";
pub const EVENT_WIDGET_COLOR_MODE: &str = "widget://color-mode";
pub const EVENT_WIDGET_DASHBOARD_DATA: &str = "widget://dashboard-data";
pub const EVENT_WIDGET_NAVIGATE: &str = "widget://navigate";
pub const EVENT_ACCESS_PATH_DIAGNOSIS_SNAPSHOT: &str = "access-path-diagnosis://snapshot";
pub const EVENT_RELEASE_PACKAGE_LOG: &str = "release-package://log";
pub const EVENT_RELEASE_PACKAGE_STATUS: &str = "release-package://status";
pub const EVENT_VAULT_LOCKED: &str = "vault://locked";
pub const EVENT_ACTION_CENTER_DISPATCH_REQUEST: &str = "action-center://dispatch-request";
pub const EVENT_REFERENCE_CARD_INIT: &str = "reference-card://init";
pub const EVENT_ACTION_CENTER_COMBINATION_RUN_UPDATED: &str =
    "action-center://combination-run-updated";

/// 供契约对账测试使用；由具名常量引用组成，无双写漂移。
#[cfg(test)]
pub const ALL: &[&str] = &[
    EVENT_MAIN_WINDOW_TOGGLE,
    EVENT_HOTKEY_NAVIGATE,
    EVENT_CLIPBOARD_CHANGED,
    EVENT_TODO_REMINDER_FIRED,
    EVENT_GLOBAL_NOTIFICATION_PUSH,
    EVENT_POMODORO_STATE_CHANGED,
    EVENT_QUICK_CAPTURE_RESET,
    EVENT_SPOTLIGHT_RESET,
    EVENT_WIDGET_COLOR_MODE,
    EVENT_WIDGET_DASHBOARD_DATA,
    EVENT_WIDGET_NAVIGATE,
    EVENT_ACCESS_PATH_DIAGNOSIS_SNAPSHOT,
    EVENT_RELEASE_PACKAGE_LOG,
    EVENT_RELEASE_PACKAGE_STATUS,
    EVENT_VAULT_LOCKED,
    EVENT_ACTION_CENTER_DISPATCH_REQUEST,
    EVENT_REFERENCE_CARD_INIT,
    EVENT_ACTION_CENTER_COMBINATION_RUN_UPDATED,
];

File: apps/desktop/src/bridge/events.ts
// 本文件被 src-tauri 契约对账测试逐行解析：常量保持 `NAME: "value",` 一行一条目。
export const APP_EVENTS = {
  /** Rust -> 主窗口：托盘/快捷键切换主窗口显隐 */
  MAIN_WINDOW_TOGGLE: "main-window-toggle",
  /** Rust -> 主窗口：命名快捷键导航 payload HotkeyNavigatePayload */
  HOTKEY_NAVIGATE: "hotkey-navigate",
  /** Rust -> 主窗口：剪贴板序列号变化 payload { sequence: number } */
  CLIPBOARD_CHANGED: "clipboard-changed",
  /** Rust -> 主窗口：Todo 提醒触发 payload ReminderDispatch 或 { refresh: true } */
  TODO_REMINDER_FIRED: "todo-reminder-fired",
  /** Rust -> 全局通知弹窗：通知队列 payload GlobalNotification[] */
  GLOBAL_NOTIFICATION_PUSH: "global-notification-push",
  /** Rust / 前端弹窗 -> 主窗口：番茄钟状态变化 payload { refresh: true } */
  POMODORO_STATE_CHANGED: "pomodoro-state-changed",
  /** Rust -> 快速捕获窗口：重置输入 */
  QUICK_CAPTURE_RESET: "quick-capture-reset",
  /** Rust -> Spotlight 窗口：重置查询 */
  SPOTLIGHT_RESET: "spotlight-reset",
  /** 仅前端：Spotlight 窗口 -> 主窗口 payload { name: string } */
  HOSTS_APPLIED: "hosts-applied",
  /** Rust -> Widget 窗口：颜色模式 payload string */
  WIDGET_COLOR_MODE: "widget://color-mode",
  /** Rust -> Widget 窗口：仪表盘数据 payload WidgetDashboardData */
  WIDGET_DASHBOARD_DATA: "widget://dashboard-data",
  /** Rust -> 主窗口：Widget 快捷导航 payload { kind: string; toolId?: string } */
  WIDGET_NAVIGATE: "widget://navigate",
  /** Rust -> 启动诊断的窗口：访问链路诊断运行快照 */
  ACCESS_PATH_DIAGNOSIS_SNAPSHOT: "access-path-diagnosis://snapshot",
  RELEASE_PACKAGE_LOG: "release-package://log",
  RELEASE_PACKAGE_STATUS: "release-package://status",
  /** Rust -> 主窗口：密码库因空闲策略进入锁定状态 */
  VAULT_LOCKED: "vault://locked",
  /** Rust -> 主窗口：动作中心打开目标工具并请求确认 */
  ACTION_CENTER_DISPATCH_REQUEST: "action-center://dispatch-request",
  /** Rust -> 参考卡窗口：初始化正文 payload { content: string } */
  REFERENCE_CARD_INIT: "reference-card://init",
  /** Rust -> 主窗口：组合动作运行状态变化，payload { runId, status } */
  ACTION_CENTER_COMBINATION_RUN_UPDATED: "action-center://combination-run-updated",
} as const;
